### PR TITLE
Added Kubernetes-the-hard-way playbook

### PR DESCRIPTION
**Pull request by ardikabs**
Related articles https://github.com/kelseyhightower/kubernetes-the-hard-way

List added roles
- calico-node (for CNI provision)
- docker (for CRI provision)
- kubernetes/master (for master provision)
- kubernetes/worker (for worker provision)

Also we structurize ansible-playgrounds project, like:
- inventory (inventory file, separated by datacenter on each directory tree)
- plays (playbook file, separated by datacenter on each directory tree)
- roles

### CODE REVIEW

File: inventory/dc1/group_vars/all [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ansible_user: CHANGE_WITH_YOUR_ANSIBLE_USER_HERE
```
Comment (dimasmamot) [inline inventory/dc1/group_vars/all:1]: Majority jenis instance di satu datacenter yang sama, punya identifier yang sama, jadi haruse ngga perlu di `CHANGE_WITH_YOUR_ANSIBLE_USER_HERE`, langsung di spesifikan ae lek jareku, soale bakal ribet pas nanti provision ansible executor baru, harus edit edit

File: inventory/dc1/group_vars/all [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+ansible_ssh_private_key_file: CHANGE_WITH_YOUR_SSH_PRIVATE_KEY_FILE
```
Comment (dimasmamot) [inline inventory/dc1/group_vars/all:2]: Iki rencana ditaro mana ? mending saranku, bikin satu directory, di level root ansible, directory `key` isinya kosong, isinya key, tapi ngga di push ke repo, ekslusif di instance ditambahnya, dan langsung di vars `all` ini di spesifikan, key yang mana yang bakal dipake, tujuan e tetep, biar seamless ga perlu edit lek clone repo ansible iki ke ansible manager baru

File: plays/dc1/k8s-bootstrap.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+---
```
Comment (dimasmamot) [inline plays/dc1/k8s-bootstrap.yml:1]: Lek iki, prasaku kok mending stage `Ensure python is present` iku langsung jadikan role `common` jadi di playbook bersih, ngga ada task tambahan

```
---
- name: Setup kubernetes bootstrap
  hosts: all
  become: yes
  vars:
    wawa: val
    qaaqa: val
    sad: [ ls1, ls2, ls3 ]

  roles:
    - common
    - role2
    - role3
```

tujuane cek modular ae, bersih playbook e, dan konfigurasi untuk ensure ensure, pre dependency iku terkontrol nde satu file, role

File: plays/dc1/swarm-bootstrap.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+---
```
Comment (dimasmamot) [inline plays/dc1/swarm-bootstrap.yml:1]: Nah koyok ngene, awamu kan nulis maneh `Ensure python is present`

mending langsung dadino role

terus panggil sesuai urutan

File: roles/calico-node/tasks/generate_certs.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+- name: Copying Certificate Authority
+  copy:
+    src: "{{ item }}"
+    dest: "{{ etcd_basedir }}/certs/"
+  with_items:
+  - "{{tls_ca}}"
+  - "{{tls_ca_key}}"
+  - "{{tls_ca_config}}"
+
```
Comment (dimasmamot) [inline roles/calico-node/tasks/generate_certs.yml:9]: `tags` jangan lupakan tags, berguna lek misale awamu pengen ngajalano beberapa `stage` nde role iki tanpa harus eksekusi semua `stage` jadi pas ansible-playbook tinggal nggae `--tags`

**Review by dimasmamot — CHANGES_REQUESTED** [review]
Ngga akeh sing isok di review, aku gapaham roles e soale wkwkkw kakean, cuma 5 iku prasaku isok lumayan oke

File: inventory/dc1/group_vars/all (hunk 2 above)
Comment (ardikabs) [reply inventory/dc1/group_vars/all:2]: jadi artinya, tiap menjalankan playbook mending di specify `ansible_user` aja ? instead ditaruh di `group_vars` ya ?

File: plays/dc1/k8s-bootstrap.yml (hunk 3 above)
Comment (ardikabs) [reply plays/dc1/k8s-bootstrap.yml:1]: okee, will be added as role

File: roles/calico-node/tasks/generate_certs.yml (hunk 5 above)
Comment (ardikabs) [reply roles/calico-node/tasks/generate_certs.yml:9]: nice idea

**Pull request merged**